tests/test_crawl.py

```diff
@@ -91,12 +91,11 @@ def test_retry_conn_failed(self):
 
     @defer.inlineCallbacks
     def test_retry_dns_error(self):
-        with mock.patch('socket.gethostbyname',
-                        side_effect=socket.gaierror(-5, 'No address associated with hostname')):
-            crawler = self.runner.create_crawler(SimpleSpider)
-            with LogCapture() as l:
-                yield crawler.crawl("http://example.com/")
-            self._assert_retried(l)
+        crawler = self.runner.create_crawler(SimpleSpider)
+        with LogCapture() as l:
+            # try to fetch the homepage of a non-existent domain
+            yield crawler.crawl("http://dns.resolution.invalid/")
```

Comment:
In the original PR (https://github.com/scrapy/scrapy/commit/877e345d79c6a14b6ab24a3f15653f6373918cfb) @dangra mentioned that `localhost666` can be resolvable under certain circumstances. Does it apply for .invalid domains as well?